src/platform/android/java/chip/platform/AndroidBleManager.java

```diff
@@ -599,9 +609,28 @@ public void onConnectionStateChange(BluetoothGatt gatt, int status, int newState
         mState = STATE_DISCOVER_SERVICE;
         gatt.discoverServices();
         return;
-      } else if (newState == BluetoothProfile.STATE_DISCONNECTED) {
+      }
+
+      if (newState == BluetoothProfile.STATE_DISCONNECTED
+          && status == 133
```

Comment:
what is the status 133  ? can we give a name to this constant?

could we 

```java
/// Unfortunately GATT 133 is very generic, however among other things
/// it may also show up on device out of range or BLE being closed due to 
/// background task termination, hence it is a candidate to retry connectivity.
private static final int kGattRetryableErrorStatus = 133;
```